test/problems/second_quantization/lattice/models/test_heisenberg_model.py

```diff
@@ -0,0 +1,103 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test HeisenbergModel."""
+
+from test import QiskitNatureTestCase
+from retworkx import PyGraph, is_isomorphic
+from qiskit_nature.second_q.properties.lattices import Lattice
+from qiskit_nature.second_q.properties import HeisenbergModel, IsingModel
+
+
+class TestHeisenbergModel(QiskitNatureTestCase):
+    """TestHeisenbergModel"""
+
+    def test_init(self):
```

Comment:
The additional tests are looking good. But I would suggest that you split this into multiple test methods. For example:
- `test_line`
- `test_triangular`
- `test_ising`
- `test_xy`
- `test_external_field`
Something along those lines.

You can also use the lattice subclasses that we have (e.g. `LineLattice`, `TriangularLattice`) instead of building out those graphs manually :+1: